From 30aa50130a7bf9cbc3a8b54032114060911b08f2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tomasz=20=C5=9Aniatowski?= <tsniatowski@vewd.com>
Date: Wed, 19 Jun 2019 20:34:46 +0200
Subject: zlib: fix gn error when is_clang=false target_cpu="arm"

Check use_thin_lto instead of thin_lto_enable_optimizations, as the
latter is only defined in a BUILD.gn so can't be checked outside of
it. Plus the intent of https://crrev.com/c/1642730 appears to have
been to error out on LTO in general anyway.

Fixes gn gen --args="is_clang=false target_cpu=\"arm\"" failing with:
ERROR at //third_party/zlib/BUILD.gn:105:17: Undefined identifier
        assert(!thin_lto_enable_optimizations,
                ^----------------------------

Bug: 819294
Change-Id: Ibbcae447b01043edc8a118f4a3a3923f2d0ae261

diff --git a/third_party/zlib/BUILD.gn b/third_party/zlib/BUILD.gn
index 3b8e5e9f746a..64a05d10c49c 100644
--- a/third_party/zlib/BUILD.gn
+++ b/third_party/zlib/BUILD.gn
@@ -2,6 +2,8 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
+import("//build/config/compiler/compiler.gni")
+
 if (current_cpu == "arm" || current_cpu == "arm64") {
   import("//build/config/arm.gni")
 }
@@ -102,7 +104,7 @@ if (use_arm_neon_optimizations) {
           assert(false, "CPU detection requires the Android NDK")
         }
       } else if (!is_win && !is_clang) {
-        assert(!thin_lto_enable_optimizations,
+        assert(!use_thin_lto,
                "ThinLTO fails mixing different module-level targets")
         cflags_c = [ "-march=armv8-a+crc" ]
       }
-- 
2.20.1

